proknow-sdk/Patient/Entities/StructureSet/StructureSetDraftLockRenewer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Text.Json;
+using System.Threading;
+using ProKnow.Exceptions;
+
+namespace ProKnow.Patient.Entities.StructureSet
+{
+    /// <summary>
+    /// Keeps a structure set lock from expiring
+    /// </summary>
+    public class StructureSetDraftLockRenewer
+    {
+        private ProKnowApi _proKnow;
+        private StructureSetItem _structureSet;
+        private Timer _timer;
+        private bool _hasStarted;
+        private TimeSpan _lockRenewalBuffer;
+
+        /// <summary>
+        /// Creates a StructureSetDraftLockRenewer
+        /// </summary>
+        /// <param name="proKnow">Root object for interfacing with the ProKnow API</param>
+        /// <param name="structureSet">The structure set</param>
+        public StructureSetDraftLockRenewer(ProKnowApi proKnow, StructureSetItem structureSet)
+        {
+            _proKnow = proKnow;
+            _structureSet = structureSet;
+            _timer = null;
+            _hasStarted = false;
+            _lockRenewalBuffer = new TimeSpan(0, 0, proKnow.LockRenewalBuffer);
+        }
+
+        /// <summary>
+        /// Starts the timer
+        /// </summary>
+        public void Start()
+        {
+            if (!_hasStarted)
+            {
+                //DateTime expiresAt = DateTime.ParseExact(_structureSet.DraftLock.ExpiresAt, "yyyy-MM-dd HH:mm:ss,fff",
+                //    CultureInfo.InvariantCulture);
+                TimeSpan expiresIn = new TimeSpan(0, 0, 0, 0, _structureSet.DraftLock.ExpiresIn);
+                //TimeSpan period = (expiresAt - DateTime.UtcNow) - _lockRenewalBuffer;

```

Comment:
Did you mean to leave in some of these commented out lines?

Reply:
No.  Removed.